File: mobile/src/main/java/treehou/se/habit/ui/settings/subsettings/general/GeneralSettingsContract.java
package treehou.se.habit.ui.settings.subsettings.general;

import treehou.se.habit.mvp.BasePresenter;
import treehou.se.habit.mvp.BaseView;

public interface GeneralSettingsContract {

    interface View extends BaseView<Presenter> {
        void updateTheme();
        void showAutoLoadSitemap(boolean show);
        void showSitemapsInMenu(Boolean show);
        void setFullscreen(boolean fullscreaan);
    }

    interface Presenter extends BasePresenter {
        void themeSelected(int theme);
        void setAutoLoadSitemap(boolean show);
        void setFullscreen(boolean fullscreaan);
        void setShowSitemapsInMenu(Boolean show);
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/settings/subsettings/general/GeneralSettingsContract.kt
package treehou.se.habit.ui.settings.subsettings.general

import treehou.se.habit.mvp.BasePresenter
import treehou.se.habit.mvp.BaseView

interface GeneralSettingsContract {

    interface View : BaseView<Presenter> {
        fun updateTheme()
        fun showAutoLoadSitemap(show: Boolean)
        fun showSitemapsInMenu(show: Boolean?)
        fun setFullscreen(fullscreen: Boolean)
    }

    interface Presenter : BasePresenter {
        fun themeSelected(theme: Int)
        fun setAutoLoadSitemap(show: Boolean)
        fun setFullscreen(fullscreen: Boolean)
        fun setShowSitemapsInMenu(show: Boolean)
    }
}
